spring-data-jdbc/src/test/java/org/springframework/data/jdbc/repository/query/QueryAnnotationH2IntegrationTests.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2018-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.jdbc.repository.query;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.data.jdbc.repository.config.EnableJdbcRepositories;
+import org.springframework.data.jdbc.testing.TestConfiguration;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+
+/**
+ * Tests the execution of queries from {@link Query} annotations on repository methods.
+ *
+ * @author Mark Paluch
+ */
+@ContextConfiguration
+@ActiveProfiles("h2")
+public class QueryAnnotationH2IntegrationTests extends AbstractQueryAnnotationIntegrationTests {
```

Comment:
The idea for `...HsqlIntegrationTests` is to have integration tests that only run for a single in memory database because the chance for it surfacing some issue between different databases is minimal. I'd therefore would like to remove this test (and the abstract super class introduced).

If you think these integration tests should run with various databases they should use the existing infrastructure and run for all supported databases.

Reply:
At least the setup is (was) not consistent with the other integration tests. If we can rewrite this test easily into a consistent form, I'm all in to follow this path.